src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,959 @@
+                            case TimingBudget.Budget20:
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x001E);
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x0022);
+                                break;
+                            case TimingBudget.Budget33:
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x0060);
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x006E);
+                                break;
+                            case TimingBudget.Budget50:
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x00AD);
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x00C6);
+                                break;
+                            case TimingBudget.Budget100:
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x01CC);
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x01EA);
+                                break;
+                            case TimingBudget.Budget200:
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x02D9);
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x02F8);
+                                break;
+                            case TimingBudget.Budget500:
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x048F);
+                                WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x04A4);
+                                break;
+                            default:
+                                throw new ArgumentException(
+                                    $"A timing budget of {TimingBudget.Budget15} can only be set in distance mode {Precision.Short}.",
+                                    nameof(value));
+                        }
+
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the precision (1 = Short, 2 = Long).
+        /// Short mode maximum distance is limited to 1.3m but results in a better ambient immunity.
+        /// Long mode can range up to 4 m in the dark with a timing budget of200 ms.
+        /// </summary>
+        public Precision Precision
+        {
+            get
+            {
+                byte temp = ReadByte((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP);
+                if (temp != 0x14 && temp != 0xA)
+                {
+                    return Precision.Unknown;
+                }
+
+                return temp == 0x14 ? Precision.Short : Precision.Long;
+            }
+            set
```

Comment:
nit, you need an extra CR between } and set.

Reply:
Done